docs/DeveloperGuide.md

```diff
@@ -2,28 +2,206 @@
+
+Given below is an example of how a user may add a student to the data base. 
+
+- Suppose the user types in the command `add wario`. The parser first splits the input into two, checking for the command and the argument.
+- It then recognises the `add` command, which then further checks if there is an argument given.
+- Since the `add` command only allows the user to add a student to the list if there were no duplicate names present before, it will run the method `findStudentByName(ArrayList<Student> masterStudentList, String name)`, to check if the name was already present in the masterlist.
+- Assuming the name "wario" did not exist in the masterlist beforehand, the parser would grant the user's request to add the student to the masterlist.
+- This `add wario` command is equivalent to having the user first type `add`, then after waiting fo the program to respond, type `wario`. In the latter scenario, the input parser deems that an argument was not present, and thus will ask the user to give the name of the student that they would like to add.
+
+All the while these commands are being processed, the runtime database `masterStudentList` is being updated, and subsequently being written to the stored database under the `data/studentInfo/Student_information.txt` file.
+
+
+### Student Details
+
+To facilitate the management of students within a tuition centre, it is imperative to have easy access and storage of important information regarding said personnel.
+
+Seen below is an UML diagram of the relevant classes dealing with storage.
+
+![DataStoringUML](./diagrams/src/DataStoring/DateStoring.png)
+
+#### Design considerations
+
+When building our implementation, we wanted our program to be able to easily access a student's details from a student object, and vice versa. 
+
+#### Implementation and rationale
+
+A `Student` is made up of a list of the subjects they are taking, along with relevant information one might need while managing their schedules and clienthood (maybe not gender, but who knows). Every `Student` object is stored in a public `static StudentList` that any class can call.
+
+Storage of these details are handled via the `StudentsAttributes` class when the user package calls upon its methods. To ensure easy access of information from either a `Student` object, and be able to retrieve the `Student` parent when having access to an `StudentAttributes` object, we linked both `Student` and `StudentAttribute`.
+
+Since both classes are referencing each other, it was easy for us to link it with the input parsing system, where we would just need a student's name, and maybe their number if there are multiple students with the same name, to be able to edit their details. 
+
+Please see the diagram below to see how identifying a student works.
+
+![studentIdentification](./diagrams/src/DataStoring/Student%20Identification.png)
```

Comment:
Shouldn't the entity names be strictly classes? Having the names as methods like findStudentByName and findStudentByNumber might not follow the sequence diagram standard.